pkg/provider/gitlab/gitlab.go

```diff
@@ -214,26 +267,58 @@ func (g *Gitlab) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretData
 	// 	"masked": true,
 	// 	"environment_scope": "*"
 	// }
-
 	var vopts *gitlab.GetProjectVariableOptions
 	if g.environment != "" {
 		vopts = &gitlab.GetProjectVariableOptions{Filter: &gitlab.VariableFilter{EnvironmentScope: g.environment}}
 	}
-	data, _, err := g.client.GetVariable(g.projectID, ref.Key, vopts)
+
+	data, resp, err := g.projectVariablesClient.GetVariable(g.projectID, ref.Key, vopts)
+	if resp.StatusCode >= 400 && resp.StatusCode != 404 && err != nil {
+		return nil, err
+	}
+
+	err = g.ResolveGroupIds()
```

Comment:
nit: could you please return the group ids here so it is immediately clear what the function does instead of changing the state of `*Gitlab`?

```go
gids, err := g.ResolveGroupIds()
```

That func returns (1) the hard-coded groupIDs from the store if `inheritFromGroups=false` or (2) the resolved groupIDs if `inheritFromGroups=true`. That may be easier to reason about and it's not needed to store too many fields on the `Gitlab` struct (just the store).

Reply:
Thanks.

The problem is that we dont have the store available for the GetSecret, GetAllSecrets calls, so I am not sure how to save much. I could use a marker instance for groupIDs in case inheritance is enabled so it does not need to be propagated into the store.
As for the reasoning, I dont think this looks much better especially since the called function returns g.groupIDs:
        groupIDs, err := g.ResolveGroupIds()
        ...
	g.groupIDs = groupIDs